### PR TITLE
Updated default search providers in KR

### PR DESCRIPTION
**Pull request by simonhong**
Resolves https://github.com/brave/brave-browser/issues/18855

Desktop & android will have Brave Search, Daum, Naver and Google as a default search provider.

### **Desktop search engine settings page**
image

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Set Region settings as South Korea
2. Launch Brave with fresh profile
3. Check search engine settings has SK specific provider list - Brave, Naver, Daum and Google
4. Check Naver is set as an initial default provider

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/onboarding/SearchEngineEnum.java [hunk 1]
```diff
@@ -24,7 +24,11 @@ public enum SearchEngineEnum {
     STARTPAGE(R.drawable.search_engine_startpage, SearchEngineEnumConstants.SEARCH_STARTPAGE_ID,
             R.string.startpage_desc),
     ECOSIA(R.drawable.ecosia, SearchEngineEnumConstants.SEARCH_ECOSIA_ID,
-            R.string.ecosia_desc);
+            R.string.ecosia_desc),
+    DAUM(R.drawable.ecosia, SearchEngineEnumConstants.SEARCH_DAUM_ID,
```
Comment (simonhong) [inline android/java/org/chromium/chrome/browser/onboarding/SearchEngineEnum.java:28]: Need daum & naver's image.

File: browser/ui/android/strings/android_brave_strings.grd [hunk 2]
```diff
@@ -1642,6 +1642,12 @@ Are you sure you want to do this?
       <message name="IDS_ECOSIA_DESC" desc="Onboarding SE text">
         Plants trees, based in Berlin
       </message>
+      <message name="IDS_DAUM_DESC" desc="Onboarding SE text">
+        Put description about daum
```
Comment (simonhong) [inline browser/ui/android/strings/android_brave_strings.grd:1646]: Need daum & naver's description

**Comment by simonhong** [comment]
@rebron @bradleyrichter Do we want to update Android onboarding with this default search provider change?
If so, I need `Naver` and `Daum` 's logo image and their description.

**Comment by rebron** [comment]
@timchilds @anthonypkeane We need getting assets for Daum and Naver, logo and description.

**Comment by anthonypkeane** [comment]
We don't show this screen in Android during onboarding. If we are, it is a bug.

**Comment by simonhong** [comment]
> We don't show this screen in Android during onboarding. If we are, it is a bug.

@anthonypkeane  This search provider onboarding is launched for first touch on omnibox.
AFAIK, it's not launched for some regions that brave search is configured as a default.
@deeppandya Am I right?

**Comment by anthonypkeane** [comment]
Slack Thread: https://bravesoftware.slack.com/archives/C01NNFM9XMM/p1656552687093119

**Comment by deeppandya** [comment]
> We don't show this screen in Android during onboarding. If we are, it is a bug. 
@anthonypkeane we still show the onboarding except countries listed here : https://github.com/brave/brave-core/blob/3598a860d23fbd1c7e98d9f562f7a57f7b7868a2/android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java#L146

**Comment by simonhong** [comment]
Kindly ping for logo and description :)

**Comment by anthonypkeane** [comment]
@rmcfadden3 can you please help with the copy for Daum & Naver?
@Sam-Sibley could you pls provide search icons for Daum & Naver?

**Comment by anthonypkeane** [comment]
Hi @simonhong 

https://github.com/brave/brave-core/pull/10590#issuecomment-1183473674

Does this work ok? Thanks

**Comment by simonhong** [comment]
> Hi @simonhong
> 
> [#10590 (comment)](https://github.com/brave/brave-core/pull/10590#issuecomment-1183473674)
> 
> Does this work ok? Thanks

@anthonypkeane Desc seems fine. and I was also waiting both providers' logo icon.
If we don't want to use different icon, I could get their official logo icon from web site.

File: browser/search_engines/search_engine_tracker.h [hunk 3]
```diff
@@ -37,7 +37,9 @@ enum class SearchEngineP3A {
   kYandex,
   kEcosia,
   kBrave,
-  kMaxValue = kBrave,
+  kDaum,
+  kNaver,
+  kMaxValue = kNaver,
```
Comment (simonhong) [inline browser/search_engines/search_engine_tracker.h:42]: cc @iefremov

**Review by deeppandya — APPROVED** [review]
Android changes LGTM

**Comment by deeppandya** [comment]
@simonhong were you able to verify the change for android on device ?

**Comment by simonhong** [comment]
> 

Yes, above screenshots came from my android phone.

File: chromium_src/components/search_engines/template_url_prepopulate_data.cc [hunk 4]
```diff
@@ -81,6 +81,13 @@ const std::vector<BravePrepopulatedEngineID> brave_engines_AU_IE = {
     PREPOPULATED_ENGINE_ID_ECOSIA,
 };
 
+const std::vector<BravePrepopulatedEngineID> brave_engines_KR = {
```
Comment (goodov) [inline chromium_src/components/search_engines/template_url_prepopulate_data.cc:84]: I wonder why these are vectors instead of plain arrays?

Comment (simonhong) [reply chromium_src/components/search_engines/template_url_prepopulate_data.cc:84]: Ok, I'll try to handle this as a f/u. Filed https://github.com/brave/brave-browser/issues/27943

**Review by goodov — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **FAILED** on `Pixel 6` running `Android 13` using the following build(s):

```
Brave | 1.49.82 Chromium: 110.0.5481.52 (Official Build) canary (32-bit)
--- | ---
Revision | 979113183ded4544a2c443aceb1629c430907e52-refs/branch-heads/5481@{#636}
OS | Android 13; Build/TQ1A.230105.002
```

#### `Test Case #2` - Upgrade from `1.49.55 Chromium: 109.0.5414.86` --> `1.49.82 Chromium: 110.0.5481.52`

**`Test Case #1`** - Default SE (Google) selected via omnibox search onboarding

* installed `1.49.55 Chromium: 109.0.5414.86` and ensured that the previous onboarding appeared once tapping on omnibox
* ensured that `Google` was selected as the default SE via the search onboarding and continued with `Google` selected
* ensured that `Google` was selected as the default SE for both `Standard Tab` & `Private Tab`

**`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|---------------|-----------------|----------------
Screenshot_20230202-023804 | Screenshot_20230202-023814 | Screenshot_20230202-023818 | Screenshot_20230202-023822

* upgraded from `1.49.55 Chromium: 109.0.5414.86` --> `1.49.82 Chromium: 110.0.5481.52`
* ensured that `Google` was still set as the default SE for both `Standard Tab` & `Private Tab`
* ensured that the new SE are being displayed within `Standard Tab` & `Private Tab`
* ensured that `Google` is also being displayed via both `Standard Tab` & `Private Tab`
* ensured that `Brave` is at the top of the list for both `Standard Tab` & `Private Tab`
* ensured that you can switch to other SE after upgrading without any issues
  * ensured that the changes are retained when restarting the browser several times

**`Example`** | **`Example`** | **`Example`**
----------------|---------------|-----------------
Screenshot_20230202-023854 | Screenshot_20230202-023858 | Screenshot_20230202-023903

**`Test Case #2`** - Selecting other SE (DDG) via omnibox search onboarding

* installed `1.49.55 Chromium: 109.0.5414.86` and ensured that the previous onboarding appeared once tapping on omnibox
* ensured that `Google` was selected as the default SE via the search onboarding
* select `DDG` as the default SE and ensure both `Standard Tab` & `Private Tab` are using `DDG`

**`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|---------------|-----------------|----------------
Screenshot_20230202-031605 | Screenshot_20230202-031614 | Screenshot_20230202-031618 | Screenshot_20230202-031622

* upgraded from `1.49.55 Chromium: 109.0.5414.86` --> `1.49.82 Chromium: 110.0.5481.52`
* ensured that `DDG` was still set as the default SE for both `Standard Tab` & `Private Tab`
* ensured that the new SE are being displayed within `Standard Tab` & `Private Tab`
* ensured that `DDG` is also being displayed via both `Standard Tab` & `Private Tab`
* ensured that `Brave` is at the top of the list for both `Standard Tab` & `Private Tab`
* ensured that you can switch to other SE after upgrading without any issues
  * ensured that the changes are retained when restarting the browser several times

**`Example`** | **`Example`** | **`Example`**
----------------|---------------|-----------------
Screenshot_20230202-031652 | Screenshot_20230202-031657 | Screenshot_20230202-031704

**`Test Case #3`** - Selecting two different SE via `Search Settings` before upgrading

* installed `1.49.55 Chromium: 109.0.5414.86` and ensured that the previous onboarding appeared once tapping on omnibox
* ensured that `Google` was selected as the default SE via the search onboarding
* select `Startpage` as the default SE and ensure both `Standard Tab` & `Private Tab` are using `Startpage`
* changed the default SE for both `Standard Tab` (Bing) & `Private Tab` (Qwant)

**`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|---------------|-----------------|----------------

**Comment by kjozwiak** [comment]
Created https://github.com/brave/brave-browser/issues/28232 which should be fixed on `Android` before uplifting the above into `1.48.x` via https://github.com/brave/brave-core/pull/16842.

**Pull request merged**